langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModel.java

```diff
@@ -0,0 +1,82 @@
+package dev.langchain4j.model.vertexai;
+
+import com.google.cloud.vertexai.VertexAI;
+import com.google.cloud.vertexai.api.Content;
+import com.google.cloud.vertexai.api.GenerateContentResponse;
+import com.google.cloud.vertexai.api.GenerationConfig;
+import com.google.cloud.vertexai.generativeai.preview.GenerativeModel;
+import com.google.cloud.vertexai.generativeai.preview.ResponseHandler;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.output.Response;
+import lombok.Builder;
+
+import java.io.IOException;
+import java.util.List;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+
+/**
+ * Represents a Google Vertex AI Gemini language model with a chat completion interface, such as gemini-pro.
+ * See details <a href="https://cloud.google.com/vertex-ai/docs/generative-ai/model-reference/gemini">here</a>.
+ */
+public class VertexAiGeminiChatModel implements ChatLanguageModel {
+
+    private final GenerationConfig generationConfig;
+    private final GenerativeModel model;
+    private final Integer maxRetries;
+
+    @Builder
+    public VertexAiGeminiChatModel(String project,
+                                   String location,
+                                   String modelName,
+                                   Float temperature,
+                                   Integer maxOutputTokens,
+                                   Integer topK,
+                                   Float topP,
+                                   Integer maxRetries) {
+
+        GenerationConfig.Builder generationConfigBuilder = GenerationConfig.newBuilder();
```

Comment:
Yes, we expect underlying SDK to do validations.
Otherwise we will be buried by all sorts of validations and will need to keep track of all the updates to them...